coresdk/src/coresdk/raspi_gpio.cpp

```diff
@@ -0,0 +1,182 @@
+// raspi_gpio.cpp
+//  splashkit
+// Created by Aditya Parmar on 20/01/2024.
+// Copyright © 2024 XQuestCode. All rights reserved.
+#include "raspi_gpio.h"
+#include "gpio_driver.h"
+#include <iostream>
+using namespace std;
+
```

Comment:
We want the `#ifdef RASPBERRY_PI` protection in this file. We can then provide a default non-pi implementation for each of these.

These functions should also check and call init if needed.

Reply:
We also need a "has_gpio" boolean function - which is just:

```cpp
bool has_gpio()
{
#ifdef RASPBERRY_PI
    return true;
#else 
    return false;
#endif
}
```